ui/src/components/credentials/IssueCredential.tsx

```diff
@@ -126,88 +146,101 @@ export function IssueCredential() {
-                            updatedValues.expirationDate?.isBefore(
-                              formData.issuanceMethod.linkExpirationDate
-                            )
-                              ? {
-                                  ...formData,
-                                  attributes: updatedValues,
-                                  issuanceMethod: {
-                                    ...formData.issuanceMethod,
-                                    linkExpirationDate: undefined,
-                                    linkExpirationTime: undefined,
-                                  },
+
+          case "issueCredential": {
+            return (
+              <Card className="issue-credential-card" title="Credential details">
+                <Space direction="vertical">
+                  <SelectSchema onSelect={setSchema} schemaID={schemaID} />
+
+                  {schema &&
+                    (() => {
+                      switch (jsonSchema.status) {
+                        case "pending":
+                        case "loading":
+                        case "reloading": {
+                          return <LoadingResult />;
+                        }
+
+                        case "failed": {
+                          return <ErrorResult error={jsonSchemaErrorToString(jsonSchema.error)} />;
+                        }
+
+                        case "successful": {
+                          return (
+                            <IssueCredentialForm
+                              initialValues={credentialFormInput.issueCredential}
+                              jsonSchema={jsonSchema.data}
+                              onBack={() => {
+                                setJsonSchema({ status: "pending" });
+                                setStep("issuanceMethod");
+                              }}
+                              onSubmit={(values) => {
+                                const updatedValues = values.expirationDate
+                                  ? {
+                                      ...values,
+                                      expirationDate: values.expirationDate.endOf("day"),
+                                    }
+                                  : values;
+                                const newCredentialFormInput: CredentialFormInput =
+                                  credentialFormInput.issuanceMethod.type === "credentialLink" &&
+                                  updatedValues.expirationDate?.isBefore(
+                                    credentialFormInput.issuanceMethod.linkExpirationDate
+                                  )
+                                    ? {
+                                        ...credentialFormInput,
+                                        issuanceMethod: {
+                                          ...credentialFormInput.issuanceMethod,
+                                          linkExpirationDate: undefined,
```

Comment:
Why are those needed? Setting them as `undefined` is equivalent to omitting them here.

Reply:
It's cleaning claimLinkExpiration if the expiration date is before linkExpirationDate. expirationDate has priority over claimLinkExpiration